File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public ArrayList<Integer> r=new ArrayList<>();

    public void helper(TreeNode root,int k){
        if(root==null || r.size()==k){
            return;
        }
        helper(root.left,k);
        this.r.add(root.val);
        helper(root.right,k);
    }

    public int kthSmallest(TreeNode root, int k) {
        this.helper(root,k);
        System.out.println(r);
        return this.r.get(k-1);
    }
}

File: Data Structures & Algorithms/insert-into-a-binary-search-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def insertIntoBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        temp=root
        def helper(r:TreeNode):
            if val>r.val:
                if r.right is not None:
                    helper(r.right)
                else:
                    r.right=TreeNode(val)
                    return
            else:
                if r.left is not None:
                    helper(r.left)
                else:
                    r.left=TreeNode(val)
                    return
        if temp is None:
            return TreeNode(val)
        helper(temp)
        return root